packages/ionic/src/commands/serve.ts

```diff
@@ -3,6 +3,7 @@ import * as chalk from 'chalk';
 import { CommandLineInputs, CommandLineOptions } from '@ionic/cli-utils';
 import { Command, CommandMetadata } from '@ionic/cli-utils/lib/command';
 import { BIND_ALL_ADDRESS, BROWSERS, DEFAULT_DEV_LOGGER_PORT, DEFAULT_LIVERELOAD_PORT, DEFAULT_SERVER_PORT } from '@ionic/cli-utils/lib/serve';
+import { Publisher } from '@ionic/discover';
```

Comment:
Please use a dynamic import in `startDevApp` to only import when `Publisher` is required.

```javascript
const { Publisher } = await import('@ionic/discover');
```